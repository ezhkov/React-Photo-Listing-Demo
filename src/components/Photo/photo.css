.photo {
    width: 30%;
    margin: 50px auto;
}
    .photo__img {
        width: 100%;
    }

    .photo-card {
        padding: 20px;
    }

    .card-body {
        
    }
    .card-body > * {
        width: 100%;
        display: block;
        text-align: center;
    }