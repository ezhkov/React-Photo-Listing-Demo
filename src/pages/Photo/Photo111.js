// import React from 'react';

// import Enzyme, { shallow, mount } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// import { Link, Route, BrowserRouter } from 'react-router-dom';
// import { LastLocationProvider } from 'react-router-last-location';
// import Photo from '.';
// import Spinner from '../../components/Spinner';

// Enzyme.configure({ adapter: new Adapter() });

// describe('Test of component of Photo', () => {
//   it('Test basis props in state', () => {
//     const props = {
//       match: {
//         params: {
//           id: 'sC-BXbi9ajw',
//         },
//       },
//     };

//     const photo = shallow(<Photo {...props} />);
//     photo.setState({
//       userName: 'User111',
//       userNic: 'url',
//       userPortfolioUrl: 'linkUrl',
//     });
//     expect(photo.find('.photo__autor-page-link').text()).toEqual("Autor's page link");
//     expect(photo.find('Link').prop('to')).toEqual("/users/url");
//     expect(photo.find('.photo__autor-link').prop('href')).toEqual("linkUrl");
//   });

//   it('Test Spinner', () => {
//     const props = {
//       match: {
//         params: {
//           id: 'sC-BXbi9ajw',
//         },
//       },
//     };

//     const photo = shallow(<Photo {...props} />);


//     photo.setState({
//       isListingLoading: false,
//     });
//     expect(photo.find(Spinner).length).toEqual(0);

//     photo.setState({
//       isListingLoading: true,
//     });

//     expect(photo.find(Spinner).length).toEqual(1);

//   });

//   it('Test Snapshot', () => {
//     const props = {
//       match: {
//         params: {
//           id: 'sC-BXbi9ajw',
//         },
//       },
//     };

//     const photo = shallow(<Photo {...props} />);
//     expect(photo).toMatchSnapshot();
//   });
// });
