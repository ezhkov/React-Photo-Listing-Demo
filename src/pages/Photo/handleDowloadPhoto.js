const handleDowloadPhoto = (...rest) => {
  let newNamePhoto = 'image.jpg';
  if (rest.length) {
    rest.every((el) => {
      if (el) {
        newNamePhoto = el[0].toUpperCase() + el.slice(1).replace(/\s/g, '-');
        return false;
      }
      return true;
    });
  }
  return newNamePhoto;
};

export default handleDowloadPhoto;
