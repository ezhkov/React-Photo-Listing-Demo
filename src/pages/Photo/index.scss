.photo {
	margin: 0px auto;
	position: relative;

	&__twitter {
		display: flex;
		align-items: center;
	}

	& .ant-card-extra,
	& .ant-card-head-title {
		padding: 10px 0 !important;
	}

	&-twitter {
		&__ava {
			display: block;
			width: 32px;
			height: 32px;
			border-radius: 100%;
			transition: all .3s;

			&:hover {
				opacity: .8;
			}
		}

		&__content {
			position: relative;
			margin-left: 5px;
			line-height: 15px;
		}

		&__user-name {
			padding: 0;
			margin: 0;
			color: #111;
			font-size: 15px;
			transition: all .3s;

			&:hover {
				opacity: .8;
			}
		}

		&__twitter-name {
			padding: 0;
			margin: 0;
			font-size: 11px;
			color: #999;
			transition: all .3s;

			&:hover {
				color: #111;
			}
		}
	}

	&-content {
		&__photo {
			position: relative;
			z-index: 1;
			display: block;
			margin: 0 auto;
			width: 300px;
			max-height: calc(100vh - 200px);
			max-width: 100%;
			min-width: 200px;
			min-height: 200px;
		}
		&__footer {
			display: flex;
			justify-content: flex-end;
			padding: 10px 10px;
		}
	}
}

