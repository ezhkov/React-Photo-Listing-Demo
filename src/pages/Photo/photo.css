.photo {
    margin: 50px auto;
    position: relative;
}
.photo__changed {
    position: relative;
    width: 100%;
    height: 100%;
}
.photo__spinner-wrap {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    z-index: 1000;
}
.photo .ant-card-body {
    text-align: center;
}
.photo__desc .ant-card-meta-title {
    font-size: 30px;
    margin-bottom: 20px;
}
.photo__desc .ant-card-meta-title::first-letter {
    text-transform: uppercase;
}
    .photo__img {
       width: 100%;
       z-index: 1;
       position: relative;
    }
    .photo__btn-go-back {
        margin-top: 10px;
    }
    .photo-card {
        padding: 20px;
       
    }
    .photo-card .view {
        cursor: pointer;
    }
    .photo-card__user-ava {
        border-radius: 100%;
        margin-right: 4px;
        width: 22px;
    }


    .card-body {
        text-align: center;
    }

    .card-title {
        text-align: center;
        display: flex;
        justify-content: center;
        align-items: center;
    }
