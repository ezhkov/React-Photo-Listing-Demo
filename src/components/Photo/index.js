import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
    Card,
    CardBody,
    CardImage,
  } from 'mdbreact';
import axios from 'axios';
import './photo.css';

class Photo extends Component {
    constructor(...args) {
      super(...args);
      this.state = {
        photoSrc: null,
        photoID: 1,
        userId: 'User',
        userName: 'UserName',
        userLastName: '',
        userPortfolioUrl: '',
        photoCreated: '',
      }
    }

    componentDidMount = () => {
        this.handlePhotoQuery();
    };
    componentDidUpdate = (prevProps, prevState) => {
        //this.handlePhotoQuery();
    };

    handlePhotoQuery = () => {
        const { match } = this.props;
        const API_URL = `https://api.unsplash.com/photos/${match.params.id}?client_id=${process.env.REACT_APP_UNSPLASH_API_KEY}`;
        
        axios.get(API_URL)
        .then((res) => {

            const photoSrc = res.data.urls.full;
            const photoID = match.params.id;
            const userId = res.data.user.id;
            const userName = res.data.user.first_name;
            const userLastName = res.data.user.last_name;
            const userPortfolioUrl = res.data.user.portfolio_url;
            const photoCreated = res.data.created_at;
  
            this.setState({
                photoSrc,
                photoID,
                userId,
                userName,
                userLastName,
                userPortfolioUrl,
                photoCreated,
            });
        })
        .catch(() => {
          console.log('api.unsplash not responding');
          
        });
    }
    render() {
        const { 
            photoSrc,
            photoID,
            userName,
            userLastName,
            userPortfolioUrl,
         } = this.state;

        return (
        <div className="photo">
            <Card className="photo-card">
                <CardImage className="img-fluid photo-card__img" src={photoSrc} />
                <CardBody>
                    <h1>Photo ID: {photoID}</h1>
                    <h2>Autor: {userName} {userLastName}</h2>
                    <a href={userPortfolioUrl}>Autor's portfolio link</a>

                </CardBody>
            </Card>
        </div>
        );
    }
}

Photo.propTypes = {
    match: PropTypes.object,
};
Photo.defaultProps = {
    match: {},
};

export default Photo;