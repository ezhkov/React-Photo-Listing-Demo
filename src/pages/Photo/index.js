import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Spinner from '../../components/Spinner';
import { Card } from "antd";
import axios from 'axios';
import './photo.css';
const { Meta } = Card;

class Photo extends Component {
    constructor(...args) {
      super(...args);
      this.state = {
        isListingLoading: false,
        photoSrc: null,
        userId: 'User',
        userName: 'UserName',
        userNic: 'UserNic',
        userLastName: '',
        userPortfolioUrl: '',
        photoCreated: '',
        photoDesc: '',
      }
    }

    componentDidMount = () => {
        this.handlePhotoQuery();
    };

    handlePhotoQuery = () => {
        const { match, history } = this.props;
        this.setState({ isListingLoading: true });
        const API_URL = `${process.env.REACT_APP_UNSPLASH_API_NAME}photos/${match.params.id}?client_id=${process.env.REACT_APP_UNSPLASH_API_KEY}`;

        axios.get(API_URL)
        .then((res) => {
            console.log('111:::', res)
            const photoSrc = res.data.urls.full;
            const userId = res.data.user.id;
            const userName = res.data.user.first_name;
            const userNic = res.data.user.username;
            const userLastName = res.data.user.last_name;
            const userPortfolioUrl = res.data.user.portfolio_url;
            const photoCreated = res.data.created_at;
            const photoDesc = res.data.description;
  
            this.setState({
                isListingLoading: false,
                photoSrc,
                userId,
                userName,
                userLastName,
                userPortfolioUrl,
                photoCreated,
                userNic,
                photoDesc,
            });
        })
        .catch((err, error) => {
          console.log('api.unsplash not responding');
          history.push('/') 
        });
    }
    render() {
        const { 
            isListingLoading,
            photoSrc,
            userPortfolioUrl,
            userNic,
            photoDesc,
         } = this.state;

        return (
        <div className="photo-container photo">
            { isListingLoading && (<Spinner />)}
            <Card
                style={{ width: '100%' }}
                cover={<img
                    className="photo__img"
                    alt="example"
                    src={photoSrc}
                    />
                }
            >
                <Meta className="photo__desc" title={`${photoDesc ? photoDesc : 'No title'}`} />
                <Link to={`/users/${userNic}`}>
                    <p>Autor's page link</p>
                </Link>
                <a className="photo__autor-link" href={userPortfolioUrl}>Autor's portfolio link</a>
            </Card>
        </div>
        );
    }
}

Photo.propTypes = {
    match: PropTypes.object,
};
Photo.defaultProps = {
    match: {},
};

export default Photo;