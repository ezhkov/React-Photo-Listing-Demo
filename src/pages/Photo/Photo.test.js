import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Link, Route } from 'react-router-dom';
import Photo from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of Photo', () => {
  it('Initial', () => { 
    const props = {
      match: {
        params: {
          id: 'sC-BXbi9ajw',
        }
      }
    }

    const photo = shallow(<Photo {...props} />);
    //photo.instance().handlePhotoQuery();
    //expect(photo.state('userNic')).toEqual('Ricardo Frantz');
    photo.setState({ userName: 'User111', userNic: 'url' });
    console.log(photo.debug());
    //expect(filter.state().showDropdown).toEqual(false);
  });
});
