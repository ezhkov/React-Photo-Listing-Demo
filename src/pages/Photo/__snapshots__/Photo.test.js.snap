// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of Photo Test Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Photo
    match={
      Object {
        "params": Object {
          "id": "sC-BXbi9ajw",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Spinner
          className="spinner"
        />,
        <Card
          cover={
            <img
              alt="example"
              className="photo__img"
              src={null}
            />
          }
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Unknown
            className="photo__desc"
            title="No title"
          />
          <Link
            replace={false}
            to="/users/UserNic"
          >
            <p
              className="photo__autor-page-link"
            >
              Autor's page link
            </p>
          </Link>
          <a
            className="photo__autor-link"
            href=""
          >
            Autor's portfolio link
          </a>
        </Card>,
      ],
      "className": "photo-container photo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "spinner",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Unknown
              className="photo__desc"
              title="No title"
            />,
            <Link
              replace={false}
              to="/users/UserNic"
            >
              <p
                className="photo__autor-page-link"
              >
                Autor's page link
              </p>
            </Link>,
            <a
              className="photo__autor-link"
              href=""
            >
              Autor's portfolio link
            </a>,
          ],
          "cover": <img
            alt="example"
            className="photo__img"
            src={null}
          />,
          "style": Object {
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "photo__desc",
              "title": "No title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <p
                className="photo__autor-page-link"
              >
                Autor's page link
              </p>,
              "replace": false,
              "to": "/users/UserNic",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Autor's page link",
                "className": "photo__autor-page-link",
              },
              "ref": null,
              "rendered": "Autor's page link",
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Autor's portfolio link",
              "className": "photo__autor-link",
              "href": "",
            },
            "ref": null,
            "rendered": "Autor's portfolio link",
            "type": "a",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Spinner
            className="spinner"
          />,
          <Card
            cover={
              <img
                alt="example"
                className="photo__img"
                src={null}
              />
            }
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <Unknown
              className="photo__desc"
              title="No title"
            />
            <Link
              replace={false}
              to="/users/UserNic"
            >
              <p
                className="photo__autor-page-link"
              >
                Autor's page link
              </p>
            </Link>
            <a
              className="photo__autor-link"
              href=""
            >
              Autor's portfolio link
            </a>
          </Card>,
        ],
        "className": "photo-container photo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "spinner",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown
                className="photo__desc"
                title="No title"
              />,
              <Link
                replace={false}
                to="/users/UserNic"
              >
                <p
                  className="photo__autor-page-link"
                >
                  Autor's page link
                </p>
              </Link>,
              <a
                className="photo__autor-link"
                href=""
              >
                Autor's portfolio link
              </a>,
            ],
            "cover": <img
              alt="example"
              className="photo__img"
              src={null}
            />,
            "style": Object {
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "photo__desc",
                "title": "No title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <p
                  className="photo__autor-page-link"
                >
                  Autor's page link
                </p>,
                "replace": false,
                "to": "/users/UserNic",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Autor's page link",
                  "className": "photo__autor-page-link",
                },
                "ref": null,
                "rendered": "Autor's page link",
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Autor's portfolio link",
                "className": "photo__autor-link",
                "href": "",
              },
              "ref": null,
              "rendered": "Autor's portfolio link",
              "type": "a",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
